### PR TITLE
📖 update release docs + fix links in issue template

### LINKED ISSUE #9849: Clearly define the CI team's scope
Related discussion: https://github.com/kubernetes-sigs/cluster-api/issues/9735

Carryover from: https://github.com/kubernetes-sigs/cluster-api/issues/9104

### PR DESCRIPTION
**Pull request by cahillsf**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9849

### CODE REVIEW

File: docs/release/role-handbooks/ci-signal/README.md [hunk 1]
```diff
@@ -22,38 +21,33 @@
...
 4. Regenerate the prowjob configuration running `make generate-test-infra-prowjobs` command from cluster-api repository. Before running this command, ensure to export the `TEST_INFRA_DIR` variable, specifying the location of the [test-infra](https://github.com/kubernetes/test-infra/) repository in your environment. For further information, refer to this [link](https://github.com/kubernetes-sigs/cluster-api/pull/9937).
 
  ```sh
   TEST_INFRA_DIR=../../k8s.io/test-infra make generate-test-infra-prowjobs
   ```
-5. Verify the jobs and dashboards a day later by taking a look at: `https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-1.6`
-6. Update `.github/workflows/weekly-security-scan.yaml` - to setup Trivy and govulncheck scanning - `.github/workflows/weekly-md-link-check.yaml` - to setup link checking in the CAPI book - and `.github/workflows/weekly-test-release.yaml` - to verify the release target is working - for the currently supported branches.
-7. Update the [PR markdown link checker](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/pr-md-link-check.yaml) accordingly (e.g. `main` -> `release-1.6`).
```
Comment (cahillsf) [inline docs/release/role-handbooks/ci-signal/README.md:50]: captured by: https://github.com/kubernetes-sigs/cluster-api/pull/11088/files#diff-f56fdb620afe282031980089c399912cfc2fe771f890d2a6e18901449cb7b968R56

**Review by chandankumar4 — APPROVED** [review]
lgtm

File: docs/release/role-handbooks/ci-signal/README.md (hunk 1 above)
Comment (sbueringer) [reply docs/release/role-handbooks/ci-signal/README.md:50]: Link doesn't work for me

Comment (sbueringer) [reply docs/release/role-handbooks/ci-signal/README.md:50]: Maybe "Update milestone applier and GitHub Actions"?

But this doesn't include PR markdown link checker as far as I can tell

**Review by hackeramitkumar — APPROVED** [review]
Small nits, as suggested by [sbueringer](https://github.com/sbueringer); 
otherwise, it looks good to me.

File: docs/release/role-handbooks/ci-signal/README.md (hunk 1 above)
Comment (cahillsf) [reply docs/release/role-handbooks/ci-signal/README.md:50]: ah, maybe its relative to the PR or i force pushed or something, sorry about that.  

yes youre right, its meant to point to the tracking bullet which links to: [update-milestone-applier-and-github-actions](https://github.com/kubernetes-sigs/cluster-api/tree/main/docs/release/role-handbooks/release-lead#update-milestone-applier-and-github-actions)

the example PR does indeed update the weekly md link check: https://github.com/kubernetes-sigs/cluster-api/pull/10357/files

unless i am missing something?

Comment (sbueringer) [reply docs/release/role-handbooks/ci-signal/README.md:50]: Yup. Point 7 is not about adding / removing branches. It's about changing the branch in https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/pr-md-link-check.yaml#L23 (it's not the weekly job it's the PR job)

Comment (cahillsf) [reply docs/release/role-handbooks/ci-signal/README.md:50]: ah i see, thanks for catching - this should be fixed

**Comment by sbueringer** [comment]
Thx!

/lgtm

Can you please just change the commit message to be a bit more expressive? :)

**Comment by fabriziopandini** [comment]
/lgtm
approve pending comment in https://github.com/kubernetes-sigs/cluster-api/pull/11088#issuecomment-2311976285

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**